core/src/main/scala/kafka/api/FetchResponse.scala

```diff
@@ -48,6 +49,36 @@ object FetchResponsePartitionData {
 
 case class FetchResponsePartitionData(error: Short = Errors.NONE.code, hw: Long = -1L, messages: MessageSet) {
   val sizeInBytes = FetchResponsePartitionData.headerSize + messages.sizeInBytes
+
+  def toMessageFormat(toMagicValue: Byte): FetchResponsePartitionData = {
```

Comment:
Read the code again. That's true. Apparently I had some wrong impression about how we process the requests...